src/net/tls.cc

```diff
@@ -887,8 +897,14 @@ class tls::reloadable_credentials_base {
             }
         }
         void do_callback(std::exception_ptr ep = {}) {
-            if (_cb && !_files.empty()) {
-                _cb(boost::copy_range<std::unordered_set<sstring>>(_files | boost::adaptors::map_keys), std::move(ep));
+            if (_cb) {
```

Comment:
If ep != nullptr, you should not do this. You potentially hide an exception (either by succeeding or by generating a new exception  - the latter more likely)